### PR TITLE
pallet: Add fn push_ink_message

### PR DESCRIPTION
**Pull request by kvinwang**
It should work. However, haven't tested it yet.

### CODE REVIEW

File: crates/phala-types/src/contract.rs [hunk 1]
```diff
@@ -26,6 +26,11 @@ pub enum CodeIndex<CodeHash> {
     WasmCode(CodeHash),
 }
 
+#[derive(Decode, Encode, TypeInfo)]
+pub enum InkCommand {
+    InkMessage { nonce: Vec<u8>, message: Vec<u8> },
```
Comment (tolak) [inline crates/phala-types/src/contract.rs:31]: What's the functionality of `nonce` here? Should we maintain it in every pallet that sends messages to ink contract?

File: pallets/phala/src/mq.rs [hunk 2]
```diff
@@ -238,12 +240,41 @@ pub mod pallet {
...
+		/// fn foo(a: u32, b: u32);
+		/// ```
+		///
+		/// Suppose it's selector is `0xdeadbeaf`. Then we can invoke this method by:
+		///
+		/// ```ignore
+		/// let a = 1_u32;
+		/// let b = 2_u32;
+		/// let selector: [u8; 4] = hex_literal::hex!("deadbeaf");
+		/// let payload = (selector, a, b).encode();
```
Comment (tolak) [inline pallets/phala/src/mq.rs:266]: That's clear enough

File: crates/phala-types/src/contract.rs (hunk 1 above)
Comment (kvinwang) [reply crates/phala-types/src/contract.rs:31]: The nonce doesn't affect the contract execution itself. But the cluster owner can config a log server to receive the ink message execution result. pRuntime would send the result with the nonce to the log server. Consider it a dev tool for contract development.

Comment (tolak) [reply crates/phala-types/src/contract.rs:31]: I am not sure if there are potential issues that exist, suppose there are multiple pallets sending messages to FatContract, unless MQ can bind the `nonce` with the sending pallet, we may get race condition of `nonce`. And if no race condition exists, we just simply increase `nonce` one by one right?

Comment (kvinwang) [reply crates/phala-types/src/contract.rs:31]: It is not necessary to be unique. You can leave it empty as this PR does.

Comment (tolak) [reply crates/phala-types/src/contract.rs:31]: Cool, LGTM then

**Review by tolak — APPROVED** [review]

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**